packages/x-date-pickers/src/CalendarPicker/DayPicker.test.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import { userEvent, screen } from '@mui/monorepo/test/utils';
+import { DayPicker } from './DayPicker';
+import { adapterToUse, createPickerRenderer } from '../../../../test/utils/pickers-utils';
+
+describe('<DayPicker />', () => {
+  const { render } = createPickerRenderer({ clock: 'fake' });
+
+  it('should set time to be midnight when selecting a date without a previous date', () => {
+    const onChange = spy();
+
+    render(
+      <DayPicker
+        date={null}
+        focusedDay={null}
+        onFocusedDayChange={() => {}}
+        onMonthSwitchingAnimationEnd={() => {}}
+        isDateDisabled={() => false}
+        isMonthSwitchingAnimating={false}
+        slideDirection="right"
+        reduceAnimations={false}
```

Comment:
Why do you define all those props? Is it for performance improvement?

Reply:
No, it's because `DayPicker` being an internal component, it was built with almost all its props required and I did not want to change its api at all.

But some default value could probably be inside `DayPicker` instead of higher in the component chain.